henryPuma/assignments/slidingwindow/lc53/lc53.js

```diff
@@ -1,19 +1,58 @@
+// Given an integer array nums, find the contiguous subarray (containing at least one number) which has the largest sum and return its sum.
+
+// Precondition: 
+// k: minimum sub array size 
+// nums: non empty array of integers
+
+// Postcondition: 
+// maxSum: integer denoting the maximum sum possible of any sub array whose length >= k
 function max_subarray_size_k(k, arr) {
-  // TODO: Write your code here
+
+  const arrLength = arr.length; // Since we're not mutating the array, we can store it's size once and use everywhere
+
+  if (arrLength < k) return 0; // When the minimum sub array required is greater than the actual array length
+
+  let maxSum = Number.MIN_VALUE; // Store the max sum overall, we initialize it to the lowest possible value
+  let windowSum = 0; // Total sum of current window when window size is == k
+
+  let startWindow = 0; // Left index of window, moves to the right when endWindow >= k
+  let endWindow = 0; // Right index of window iterates forward throughout the array
+
+  const minIndex = k-1; // Compute k-1 once and use it to compare with endWindow for every iteration in the while loop on line 22
+
+  while(endWindow < arrLength){
+    windowSum += arr[endWindow]; // Accumulate the sum of our current window
+
+    if(endWindow >= minIndex ){ // Check if endWindow is at index where index >= k-1
+      // If true, then windowSum has the current sum of our window where windowLength == k
+      maxSum = windowSum > maxSum ? windowSum : maxSum; // Update max sum
+      windowSum -= arr[startWindow]; // Remove value at startWindow from windowSum since we are shifting the startWindow index to the right
+      startWindow++; // Shift startWindow to the right
+    }
+    endWindow++; // Shift endWindow to the right
+  }
```

Comment:
I like the use of high level overview! #goodWork

I will share that many interviewers would prefer to see the high level overview as one big blob above the code base. This is the one feedback I will give at this time, because having the comments in-line makes the code a bit harder to read through!

Second feedback is: Where is the love for typing it twice?!? Make sure to practice always typing your solution twice as this builds muscle memory and memory access.

#GRIND #